util/reggen/field.py

```diff
@@ -125,17 +125,14 @@ def from_raw(reg_name: str,
         else:
             hwaccess = default_hwaccess
 
-        raw_hwqe = rd.get('hwqe')
-        if raw_hwqe is not None:
-            if hwext:
-                raise ValueError('The {} register has hwext set, but field {} '
-                                 'has a hwqe field-specific value specified, '
-                                 'which is illegal when combined with hwext.'
-                                 .format(reg_name,
-                                         name))
-            hwqe = check_bool(raw_hwqe, 'hwqe field for {}'.format(where))
-        else:
-            hwqe = default_hwqe
+        if hwext and ('hwqe' in rd):
+            raise ValueError('The {} register has hwext set, but field {} has a'
+                             'a hwqe field-specific value specified, which is '
+                             'illegal when combined with hwext.'
+                             .format(reg_name,
+                                     name))
```

Comment:
I'm not sure that this is quite right. What happens if `rd` doesn't contain `hwqe` but `hwext` is true and so is `default_hwqe`? I would have thought you could do something much simpler:
```python3
raw_hwqe = rd.get('hwqe', default_hwqe)
hwqe = check_bool(raw_hwqe, 'hwqe field for {}'.format(where))
if hwext and hwqe:
    raise ValueError('oh noes...')
```
It doesn't explain exactly where the offending configuration came from, but I'm not sure it really matters.

Also, could you move this patch to the previous commit?

Reply:
Let me see if I can handle the exception in a way that doesn't require duplicating code across both register.py and field.py. Right now, I suppose the logic is funky with this code just attempting to defer the hwext/hwqe handling to where the code is now.